Source/DafnyCore/AST/Grammar/Printer/Printer.Expression.cs

```diff
@@ -1193,6 +1193,28 @@ void PrintCasePattern<VT>(CasePattern<VT> pat)
         wr.Write("[BoogieFunctionCall]");  // this prevents debugger watch window crash
       } else if (expr is Resolver_IdentifierExpr) {
         wr.Write("[Resolver_IdentifierExpr]");  // we can get here in the middle of a debugging session
+      } else if (expr is DecreasesToExpr decreasesToExpr) {
```

Comment:
Out of scope: Looking forward to us setting up a framework so new AST types can group all their code into a single location. Also, I think a single grammar definition should define both the parser and printer.

Did you add a test for this printer? Seems like a test that does parse->printToA->parseFromA->printToB->diffAagainstB that takes a Dafny program containing all AST node types, would be good to have.

Reply:
> Out of scope: Looking forward to us setting up a framework so new AST types can group all their code into a single location. Also, I think a single grammar definition should define both the parser and printer.

I agree!

> Did you add a test for this printer? Seems like a test that does parse->printToA->parseFromA->printToB->diffAagainstB that takes a Dafny program containing all AST node types, would be good to have.

The tests I've added do test this printer. And, yes, I think it would be great to have such a test for the entire parser and pretty-printer.